ts/demo-functions/src/helm_template.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { safeLoadAll } from 'js-yaml';
+import {
+  Configs,
+  FunctionConfigError,
+  isKubernetesObject,
+  generalResult,
+} from 'kpt-functions';
+import { spawnSync } from 'child_process';
+import { isConfigMap } from './gen/io.k8s.api.core.v1';
+
+const CHART_NAME = 'name';
+const CHART_PATH = 'chart_path';
+const VALUES_PATH = '--values';
+
+// Render chart templates locally using helm template.
+export async function helmTemplate(configs: Configs) {
+  // Validate config data and read arguments.
+  const args = readArguments(configs);
+  args.unshift('template');
+
+  let error;
+  try {
+    const child = spawnSync('helm', args);
+    error = child.stderr;
+    let objects = safeLoadAll(child.stdout);
+    objects = objects.filter(o => isKubernetesObject(o));
+    configs.insert(...objects);
+  } catch (err) {
+    configs.addResults(generalResult(err, 'error'));
+  }
+  if (error && error.length > 0) {
+    configs.addResults(
+      generalResult(`Helm template command results in error: ${error}`, 'error')
+    );
+  }
+}
+
+function readArguments(configs: Configs) {
+  const args: string[] = [];
+
+  // Helm template expects name then chart path then remaining flags
+  const name = configs.getFunctionConfigValue(CHART_NAME);
+  if (name) {
+    args.push(name);
+  }
+  const chartPath = configs.getFunctionConfigValue(CHART_PATH);
+  if (chartPath) {
+    args.push(chartPath);
+  }
+
+  // Remaining flags
+  const data = readConfigDataOrThrow(configs);
```

Comment:
There's a lot of complexity in this readArgumetns, aren't' you just getting 3 fields in data?

Even if want to have the flexibility to get arbitrary flags (which should be documented in `usage`), probably want do:

1. Get all the values from `data`.
2. Reorder them the way Helm expects (Bring CHART_NAME and CHART_PATH to front)

Reply:
Done

Reply:
Not done?